src/item_collaborative_filtering.py

```diff
@@ -23,53 +23,98 @@ def __generate_item_pairs(self, df, item):
         :return: list of tuples, length=df['item'].nunique()
         """
         if item is not None:
-            return [(item, paired_item) for paired_item in df[self.item_column].unique() if paired_item!=item]
+            return [(item, paired_item) for paired_item in df[self.item_column].unique() if paired_item != item]
         else:
             return [(item, paired_item) for
                     item, paired_item in permutations(df[self.item_column].unique(), 2)]
 
+    def __item_users(self, df):
+        """
+        Looks up the users that interacted (e.g. watched, purchased) with an item
+        :param df: dataframe with columns [user_id, item_id]
+        :return: dictionary {item: set(user1, user2, ...)}
```

Comment:
Does it really return it though? From the code it seems like the method doesn't return anything, it just sets a variable :)